client/ayon_nuke/api/lib.py

```diff
@@ -1545,19 +1547,66 @@ def set_viewers_colorspace(self, imageio_nuke):
                 "It had wrong color profile".format(erased_viewers))
 
     def _display_and_view_formatted(self, view_profile):
-        """ Format display and view profile string
+        """ Format display and view profile.
+
+        Gets all possible display and view colorspaces and tries to set
+        viewerProcess to the one that is found in settings. It is iterating
+        over all possible combinations of display and view colorspaces. Those
+        could be separated by color value separator defined in constants.
 
         Args:
             view_profile (dict): view and display profile
 
         Returns:
             str: formatted display and view profile string
         """
-        display_view = create_viewer_profile_string(
-            view_profile["view"], view_profile["display"], path_like=False
-        )
-        # format any template tokens used in the string
-        return StringTemplate(display_view).format_strict(self.formatting_data)
+        # default values
+        views = [view_profile["view"]]
+        displays = [view_profile["display"]]
+
+        # separate all values by path separator
+        if COLOR_VALUE_SEPARATOR in view_profile["view"]:
+            views = view_profile["view"].split(
+                COLOR_VALUE_SEPARATOR)
+        if COLOR_VALUE_SEPARATOR in view_profile["display"]:
+            displays = view_profile["display"].split(
+                COLOR_VALUE_SEPARATOR)
+
+        # generate all possible combination of display/view
+        display_views = []
+        for view in views:
+            for display in displays:
+                display_views.append(
+                    create_viewer_profile_string(
+                        view.strip(), display.strip(), path_like=False
+                    )
+                )
+
+        for dv_item in display_views:
+            log.debug("Trying to set viewerProcess: `{}`".format(dv_item))
+            # format any template tokens used in the string
+            dv_item_resolved = StringTemplate(dv_item).format_strict(
+                self.formatting_data
+            )
+            log.info("Resolved viewerProcess: `{}`".format(dv_item_resolved))
+
+            if dv_item_resolved in self.get_display_and_view_colorspaces():
+                return dv_item_resolved
```

Comment:
I would do it in the loop.
```suggestion
        dv_colorspaces = self.get_display_and_view_colorspaces()
        for view in views:
            for display in displays:
                dv_item = create_viewer_profile_string(
                    view.strip(), display.strip(), path_like=False
                )

                log.debug("Trying to set viewerProcess: `{}`".format(dv_item))
                # format any template tokens used in the string
                dv_item_resolved = StringTemplate(dv_item).format_strict(
                    self.formatting_data
                )
                log.info(
                    "Resolved viewerProcess: `{}`".format(dv_item_resolved)
                )

                if dv_item_resolved in dv_colorspaces:
                    return dv_item_resolved
```


If the content of the loop should be separated then I would use submethod:
```python
        ...
        dv_colorspaces = self.get_display_and_view_colorspaces()
        for view in views:
            for display in displays:
                dv_item_resolved = self._get_resolved_display_view_item(
                    view, display, dv_colorspaces
                )
                if dv_item_resolved:
                    return dv_item_resolved

    def _get_resolved_display_view_item(
        self, view, display, available_colorspaces
    ):
        dv_item = create_viewer_profile_string(
            view.strip(), display.strip(), path_like=False
        )

        log.debug("Trying to set viewerProcess: `{}`".format(dv_item))
        # format any template tokens used in the string
        dv_item_resolved = StringTemplate(dv_item).format_strict(
            self.formatting_data
        )
        log.info(
            "Resolved viewerProcess: `{}`".format(dv_item_resolved)
        )

        if dv_item_resolved in available_colorspaces:
            return dv_item_resolved
        return None
```

Reply:
Thank you for the suggestions. I was refactoring the code a little and this wouldn't apply anymore. Thanks again!